File: skills/optimize-c-code/evals/files/stats/stats.c
#include "stats.h"

static int count;

double mean(const double *values,int count){
  double total=0;
  for(int i=0;i<count;i++) total+=values[i];
  return count?total/count:0;
}

double variance(const double*values, int n)
{
    double total = 0; double m = mean(values, n);
    for (int i = 0; i < n; i++) { double d = values[i] - m; total += d*d; }
  return n ? total/n : 0;
}

int count_above(const double *values, int n, double limit)
{
    int count = 0;
    for (int i = 0; i < n; i++)
        if (values[i] > limit)
            count++;
    return count;
}

int record(double value)
{
        (void)value;
    return ++count;
}

File: skills/optimize-swift-code/evals/files/stats/Sources/Stats/Stats.swift
/// Running totals shared by every ingestion task.
public actor Stats {
    public private(set) var total = 0
    public private(set) var count = 0

    public init() {}

    public func add(_ value: Int) {
        total += value
        count += 1
    }
}

/// Called by each ingestion task with a batch of about 100k values.
public func record(_ values: [Int], into stats: Stats) async {
    for value in values {
        await stats.add(value)
    }
}
